G-API: ARMv7 build is broken

[Nightly build](http://pullrequest.opencv.org/buildbot/builders/master_ARMv7-lin/builds/10967).

relates #15216

Error message:
```
In file included from /build/precommit_armv7/opencv/modules/gapi/src/backends/ocl/goclbackend.hpp:18:0,
                 from /build/precommit_armv7/opencv/modules/gapi/src/backends/ocl/goclbackend.cpp:25:
/build/precommit_armv7/opencv/modules/gapi/include/opencv2/gapi/ocl/goclkernel.hpp: In constructor 'cv::detail::tracked_cv_umat::tracked_cv_umat(cv::UMat&)':
/build/precommit_armv7/opencv/modules/gapi/include/opencv2/gapi/ocl/goclkernel.hpp:122:69: error: invalid initialization of non-const reference of type 'cv::UMat&' from an rvalue of type '<brace-enclosed initializer list>'
     tracked_cv_umat(cv::UMat& m) : r{ (m) }, original_data{ nullptr } {}
                                                                     ^
```

and warning:
```
/build/master_ARMv7-lin/opencv/modules/gapi/src/executor/gstreamingexecutor.cpp: In member function 'void cv::gimpl::GStreamingExecutor::setSource(cv::GRunArgs&&)':
/build/master_ARMv7-lin/opencv/modules/gapi/src/executor/gstreamingexecutor.cpp:549:22: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     if (num_videos > 1u)
                      ^
```

--- a/modules/gapi/include/opencv2/gapi/ocl/goclkernel.hpp
+++ b/modules/gapi/include/opencv2/gapi/ocl/goclkernel.hpp
@@ -119,7 +119,7 @@ template<class T> struct ocl_get_in
 struct tracked_cv_umat{
     //TODO Think if T - API could reallocate UMat to a proper size - how do we handle this ?
     //tracked_cv_umat(cv::UMat& m) : r{(m)}, original_data{m.getMat(ACCESS_RW).data} {}
-    tracked_cv_umat(cv::UMat& m) : r{ (m) }, original_data{ nullptr } {}
+    tracked_cv_umat(cv::UMat& m) : r(m), original_data{ nullptr } {}
     cv::UMat &r; // FIXME: It was a value (not a reference) before.
                  // Actually OCL backend should allocate its internal data!
     uchar* original_data;

--- a/modules/gapi/src/executor/gstreamingexecutor.cpp
+++ b/modules/gapi/src/executor/gstreamingexecutor.cpp
@@ -546,7 +546,7 @@ void cv::gimpl::GStreamingExecutor::setSource(GRunArgs &&ins)
         return util::holds_alternative<cv::gapi::wip::IStreamSource::Ptr>(arg);
     };
     const auto num_videos = std::count_if(ins.begin(), ins.end(), is_video);
-    if (num_videos > 1u)
+    if (num_videos > 1)
     {
         // See below why (another reason - no documented behavior
         // on handling videos streams of different length)

